### ISSUE TITLE
sklearn-compatible API: add fit(), score(), get_params(), set_params()

### ISSUE THREAD

**Issue by direkkakkar319-ops** [skeval-ai/skeval#16]
## Summary
Make `SentenceClassifier` a proper sklearn-compatible estimator so it can be used inside `sklearn.pipeline.Pipeline` and `GridSearchCV`.

## Changes required
- Rename `train()` → `fit(X, y)` as the primary method
- Keep `train()` as a deprecated alias that emits `DeprecationWarning` (remove in v0.3.0)
- Add `score(X, y)` → returns accuracy directly
- Add `get_params(deep=True)` and `set_params(**params)`
- Implement `__sklearn_tags__()` for sklearn 1.6+ compatibility
- Pass `sklearn.utils.estimator_checks.check_estimator(SentenceClassifier())`

## Target API
```python
clf = SentenceClassifier(embed_dim=64, epochs=20, random_state=42)
clf.fit(X_train, y_train)
clf.predict(X_test)
clf.score(X_test, y_test)
clf.get_params()  # {'embed_dim': 64, 'epochs': 20, ...}
```

## Acceptance criteria
- `SentenceClassifier` works inside `sklearn.pipeline.Pipeline` end-to-end
- `check_estimator()` passes in CI

**Milestone:** v0.2.0
